tools/serviceTest.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+########################### TESTING ############################
+# For now on Mac:
+# docker logs -f `docker ps -q --filter name=$SERVICE_NAME`
+#
+# For now on Linux:
+# tail -f /var/log/syslog
+################################################################
+
+# $1 == $SERVICE_NAME
+# $2 == "key" to know if service is successfully runnning. If found, exit(0)
+# $3 == timeout - if exceeded, service failed. exit(1)
+
+name=$1
+match=$2
+timeOut=$3
+START=$SECONDS
+
+##################################### Check the operating system #########################################
+if [ $(uname -s) == "Darwin" ]; then
+    # This is a MAC machine
+    docker logs -f `docker ps -q --filter name=$name` > output.txt &
+else
+    # This is a LINUX machine
+    tail -f /var/log/syslog > output.txt &
+fi
+
+####################### Loop until until either MATCH is found or TIMEOUT is exceeded #####################
+while :
```

Comment:
Instead of running the log cmd in the background to send its output to a file, you should just pipe it into the while loop like:
```
$cmd | while read line; do  
  # something
done
```

This avoids a couple problems in your current implementation:
1. you don't kill the background job at the end
1. you delete output.txt while the log cmd is still writing to it